src/mixer/basetrackplayer.cpp

```diff
@@ -309,6 +311,20 @@ void BaseTrackPlayerImpl::slotEjectTrack(double v) {
     if (v <= 0) {
         return;
     }
+
+    mixxx::Duration elapsed = m_ejectTimer.restart();
+
+    // Double-click always restores the last replaced track, i.e. un-eject the second
+    // last track: the first click ejects or unejects, and the second click reloads.
+    if (elapsed < mixxx::Duration::fromSeconds(0.5)) {
+        TrackPointer lastEjected = m_pPlayerManager->getSecondLastEjectedTrack();
```

Comment:
The crash happens in the unrelated EngineSyncTest.EjectTrackSyncRemains test because m_pPlayerManager is null in that case.

Reply:
oh, thanks for the heads-up